quarkus/service/src/main/java/org/apache/polaris/service/quarkus/metrics/QuarkusValueExpressionResolver.java

```diff
@@ -22,17 +22,23 @@
 import io.micrometer.common.lang.Nullable;
 import jakarta.annotation.Nonnull;
 import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
 import org.apache.polaris.core.context.RealmContext;
 
 @ApplicationScoped
 public class QuarkusValueExpressionResolver implements ValueExpressionResolver {
 
+  @Inject QuarkusMetricsConfiguration metricsConfiguration;
+
   @Override
   public String resolve(@Nonnull String expression, @Nullable Object parameter) {
     // TODO maybe replace with CEL of some expression engine and make this more generic
-    if (parameter instanceof RealmContext realmContext && expression.equals("realmIdentifier")) {
+    if (metricsConfiguration.realmIdTag().apiMetricsEnabled()
+        && parameter instanceof RealmContext realmContext
+        && expression.equals("realmIdentifier")) {
       return realmContext.getRealmIdentifier();
     }
-    return null;
+    // FIXME cannot return null here, see https://github.com/quarkusio/quarkus/issues/47891
+    return "";
```

Comment:
```suggestion
    return "???";
```
... to indicate that it's "unknown"? (IIRC there was a similar usage of `???` elsewhere)

Reply:
Not sure. I wasn't able to 100% verify this, but it seems that the empty string in a tag value is treated as not tag at all. Also the `???` is going to go away soon (fix coming in Quarkus 3.24).

Reply:
WFM